File: VB/WebSite/App_Code/PersistentObjects.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls
Imports DevExpress.Xpo

Namespace MasterDetailWithXPO
	Public Class Customer
		Inherits XPObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		Private _Name As String
		Public Property Name() As String
			Get
				Return _Name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _Name, value)
			End Set
		End Property

		<Association("Customer-Orders")> _
		Public ReadOnly Property Orders() As XPCollection(Of Order)
			Get
				Return GetCollection(Of Order)("Orders")
			End Get
		End Property
	End Class
	Public Class Order
		Inherits XPObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		Private _Customer As Customer
		<Association("Customer-Orders")> _
		Public Property Customer() As Customer
			Get
				Return _Customer
			End Get
			Set(ByVal value As Customer)
				SetPropertyValue("Customer", _Customer, value)
			End Set
		End Property
		Private _Date As DateTime
		Public Property [Date]() As DateTime
			Get
				Return _Date
			End Get
			Set(ByVal value As DateTime)
				SetPropertyValue("Date", _Date, value)
			End Set
		End Property
		Private _Totals As Decimal
		Public Property Totals() As Decimal
			Get
				Return _Totals
			End Get
			Set(ByVal value As Decimal)
				SetPropertyValue("Totals", _Totals, value)
			End Set
		End Property

	End Class
End Namespace

File: CS/WebSite/App_Code/PersistentObjects.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using DevExpress.Xpo;

namespace MasterDetailWithXPO {
    public class Customer: XPObject {
        public Customer(Session session)
            : base(session) { }

        private string _Name;
        public string Name {
            get { return _Name; }
            set { SetPropertyValue("Name", ref _Name, value); }
        }

        [Association("Customer-Orders")]
        public XPCollection<Order> Orders {
            get { return GetCollection<Order>("Orders"); }
        }
    }
    public class Order: XPObject {
        public Order(Session session)
            : base(session) { }

        private Customer _Customer;
        [Association("Customer-Orders")]
        public Customer Customer {
            get { return _Customer; }
            set { SetPropertyValue("Customer", ref _Customer, value); }
        }
        private DateTime _Date;
        public DateTime Date {
            get { return _Date; }
            set { SetPropertyValue("Date", ref _Date, value); }
        }
        private decimal _Totals;
        public decimal Totals {
            get { return _Totals; }
            set { SetPropertyValue("Totals", ref _Totals, value); }
        }
        
    }
}
